keras/src/ops/core.py

```diff
@@ -1165,3 +1165,30 @@ def grad(*args, upstream):
     ```
     """
     return backend.core.custom_gradient(f)
+
+
+class Remat(Operation):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, func, *args, **kwargs):
+        remat_func = backend.core.remat(func)
+        return remat_func(*args, **kwargs)
+
+    def compute_output_spec(self, f, *args, **kwargs):
+        return backend.compute_output_spec(f, *args, **kwargs)
+
+
+@keras_export("keras.ops.remat")
+def remat(f, *args, mode=None, output_size=None, layer_names=None, **kwargs):
+    """
```

Comment:
Follow other ops here for example of docstrings.
- Always start the one liner immediately after triple quote.
- Add some usage examples.